opendbc/car/psa/values.py

```diff
@@ -0,0 +1,82 @@
+from dataclasses import dataclass, field
+
+from opendbc.car.structs import CarParams
+from opendbc.car import AngleSteeringLimits, Bus, CarSpecs, DbcDict, PlatformConfig, Platforms, uds
+from opendbc.car.docs_definitions import CarDocs, CarHarness, CarParts
+from opendbc.car.fw_query_definitions import FwQueryConfig, Request
+
+Ecu = CarParams.Ecu
+
+class CarControllerParams:
+  STEER_STEP = 1
+
+  ANGLE_LIMITS: AngleSteeringLimits = AngleSteeringLimits(
+    390, # deg
+    ([0., 5., 25.], [2.5, 1.5, 0.2]),
+    ([0., 5., 25.], [5., 2.0, 0.3]),
+  )
+  STEER_DRIVER_ALLOWANCE = 10  # Driver intervention threshold, 1.0 Nm
+
+  def __init__(self, CP):
+    pass
+
+@dataclass(frozen=True, kw_only=True)
+class PSACarSpecs(CarSpecs):
+  tireStiffnessFactor: float = 1.03
+
+@dataclass
+class PSACarDocs(CarDocs):
+  package: str = "Adaptive Cruise Control (ACC) & Lane Assist"
+  car_parts: CarParts = field(default_factory=CarParts.common([CarHarness.psa_a]))
+
+@dataclass
+class PSAPlatformConfig(PlatformConfig):
+  dbc_dict: DbcDict = field(default_factory=lambda: {
+    Bus.pt: 'psa_aee2010_r3',
+  })
+
+class CAR(Platforms):
+  PSA_PEUGEOT_208 = PSAPlatformConfig(
+    [PSACarDocs("Peugeot 208")],
+    PSACarSpecs(
+      mass=1530, # electric variant
+      wheelbase=2.540,
+      steerRatio=17.6,
+      centerToFrontRatio=0.44,
+    ),
+  )
+
+PSA_DIAG_REQ  = bytes([uds.SERVICE_TYPE.DIAGNOSTIC_SESSION_CONTROL, 0x01])
+PSA_DIAG_RESP = bytes([uds.SERVICE_TYPE.DIAGNOSTIC_SESSION_CONTROL + 0x40, 0x01])
+
+PSA_SERIAL_REQ = bytes([uds.SERVICE_TYPE.READ_DATA_BY_IDENTIFIER,  0xF1, 0x8C])
+PSA_SERIAL_RESP = bytes([uds.SERVICE_TYPE.READ_DATA_BY_IDENTIFIER + 0x40, 0xF1, 0x8C])
+
+PSA_VERSION_REQ  = bytes([uds.SERVICE_TYPE.READ_DATA_BY_IDENTIFIER, 0xF0, 0xFE])
+PSA_VERSION_RESP = bytes([uds.SERVICE_TYPE.READ_DATA_BY_IDENTIFIER + 0x40, 0xF0, 0xFE])
+
+PSA_RX_OFFSET = -0x20
+
+# PSA ECUs send an unpadded response on tester_present request.
+# _is_tester_present_response only accepts padded responses, so needed to be modified.
+
+FW_QUERY_CONFIG = FwQueryConfig(
+  requests=[request for bus in (0, 1) for request in [
+    Request(
+      [PSA_DIAG_REQ, PSA_SERIAL_REQ],
+      [PSA_DIAG_RESP, PSA_SERIAL_RESP],
+      rx_offset=PSA_RX_OFFSET,
+      bus=bus,
+      obd_multiplexing=False,
+    ),
+    Request(
+      [PSA_DIAG_REQ, PSA_VERSION_REQ],
+      [PSA_DIAG_RESP, PSA_VERSION_RESP],
+      rx_offset=PSA_RX_OFFSET,
+      bus=bus,
+      obd_multiplexing=False,
+    ),
+  ]]
```

Comment:
Are both requests necessary? It might not be merged as-is if this would increase the total FW query time without a good reason

If it needs a bit more time to understand that then maybe all of the FW query changes could be in a follow-up PR